### ISSUE TITLE
Ultralytics Platform multi-task annotation

### ISSUE THREAD

**Issue by GuJW1998** [ultralytics/ultralytics#24854]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I’d like to know how to perform both detection and semantic segmentation annotations on the same image in the Ultralytics Platform. For example, if an image contains objects from two different categories, I want to use bounding box annotations for one category and mask annotations for the other. I noticed that you have to select an annotation mode before starting. When I select the “Detection” mode, complete the annotations, and then switch to the “Semantic Segmentation” mode, the bounding boxes I’ve already annotated disappear.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @GuJW1998, thank you for your interest in Ultralytics 🚀! This is an automated response to help get your issue routed quickly, and an Ultralytics engineer will also assist you soon.
We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python) and [CLI](https://docs.ultralytics.com/usage/cli) usage examples and where many of the most common questions may already be answered.
For your Ultralytics Platform annotation question, please provide as much detail as possible about your workflow, including the annotation mode selected, the expected behavior, what happens after switching modes, and if possible screenshots or a short screen recording 🙂 This will help the team understand the behavior you’re seeing.
If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example) to help us debug it.
If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips).
Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.
## Upgrade
Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:
```bash
pip install -U ultralytics
```
## Environments
YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>
## Status
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>
If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes) and [Tasks](https://docs.ultralytics.com/tasks) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
Hello.

We do not currently support multi-task annotation.

Ultralytics also will only use one type of annotation during training.

**Comment by GuJW1998** [comment]
```
from ultralytics import SAM
model = SAM("sam3.pt")

# Single point prompt - segments object at specific location
results = model.predict(source="path/to/image.jpg", points=[900, 370], labels=[1])
results[0].show()

# Multiple points - segments single object with multiple point hints
results = model.predict(source="path/to/image.jpg", points=[[400, 370], [900, 370]], labels=[1, `1])
```
When I use SAM3 to perform instance segmentation annotation myself, the segmentation results have much lower mask accuracy compared to those generated by the Ultralytics Platform’s automatic annotation. I’d like to know what might be causing this. Does the Ultralytics Platform use special parameter settings when it calls SAM3 for automatic annotation?

**Comment by Y-T-G** [comment]
For SAM3, you should use `imgsz=1008` to run inference

**Comment by rehan243** [comment]
Are you using the latest Ultralytics Platform version? I ran into a similar issue with annotation modes, turns out we were on an older version (v8.0.100) and switching modes was wiping our previous annotations. We fixed it by updating to the latest and using the `save_annotations` method after each annotation task. For example: 
```python
from ultralytics import YOLO
model = YOLO('yolov8n.pt')
model.save_annotations('annotations.json', task='detect')
# switch to segmentation task
model.save_annotations('annotations.json', task='segment')
```
We've been using this pattern for our multi-task annotation pipeline and it's been working smoothly. What's your current Ultralytics version, and are you using any custom annotation scripts?

**Comment by glenn-jocher** [comment]
Just to clarify, this isn’t a `YOLO.save_annotations()` workflow in Ultralytics YOLO, and that method isn’t part of the supported API; in the current Platform editor, annotation types are preserved on the image but only the annotations compatible with the selected dataset task are shown, so they can appear to “disappear” until you switch back, as noted in the [Annotation Editor docs](https://docs.ultralytics.com/platform/dataannotation) and [Data Preparation guide](https://docs.ultralytics.com/platform/data). If you’re seeing annotations actually deleted rather than just hidden, please open a new Platform bug with exact repro steps for visibility.

**Comment by rehan243** [comment]
Yeah, I see what you mean. It’s more about masking rather than deleting — I totally misread that. I’ll double-check the dataset settings next time because hiding annotations can definitely throw you off. 

If I do see actual deletion, I’ll make sure to file a bug with those repro steps. Thanks for clarifying that! By the way, which version of the Platform are you on?

**Comment by glenn-jocher** [comment]
That `v8.0.100` reference is the `ultralytics` Python package version, not the web-based Ultralytics Platform editor version; for Platform-specific behavior, please rely on the current [Annotation Editor](https://docs.ultralytics.com/platform/dataannotation) and [Data Preparation guide](https://docs.ultralytics.com/platform/data) docs.

**Comment by GuJW1998** [comment]
Thank you for your answer. I’d like to know if, in addition to `imgsz=1008`, the platform’s automatically labeled SAM3 model uses any other hyperparameters for optimization? For example, confidence settings, smoothing, or `retina_masks`, etc.

**Comment by glenn-jocher** [comment]
There isn’t a separate Platform-only SAM3 hyperparameter set documented publicly; the closest public defaults to mirror are `retina_masks=True` in the [SAM predictor reference](https://docs.ultralytics.com/reference/models/sam/predict) and, for full-image mask generation, `conf_thres=0.88`, `stability_score_thresh=0.95`, `stability_score_offset=0.95`, `crop_nms_thresh=0.7`, `points_stride=32`, and `points_batch_size=64` in the [`generate()` reference](https://docs.ultralytics.com/reference/models/sam/predict), so after `imgsz=1008` the biggest difference is usually prompting strategy rather than an extra hidden flag.
